pinot-spi/src/main/java/org/apache/pinot/spi/config/table/UpsertConfig.java

```diff
@@ -94,6 +94,9 @@ public enum ConsistencyMode {
   @JsonPropertyDescription("Whether to drop out-of-order record")
   private boolean _dropOutOfOrderRecord;
 
+  @JsonPropertyDescription("Whether to pause partial upsert table's partition ingestion during commit")
+  private boolean _pausePartialUpsertPartitionIngestionDuringCommit;
```

Comment:
I think we should pause consumption by default. Consider renaming it to `allowPartialUpsertConsumptionDuringCommit` and reverse the behavior?

Reply:
cc @tibrewalpratik17 What's your opinion on enabling the new feature by default?

Reply:
+1 on making this the default behaviour! 
Would also suggest to add a cluster-level and table-level config to enable / disable it.

Reply:
Making default as enabled sounds good. This is a behaviour change, I also think it makes sense to make this a cluster level property as well so it can be toggled at bulk.